src/sentry/models/authprovider.py

```diff
@@ -48,5 +61,87 @@ def get_provider(self):
+            return SentryAppInstallationForProvider.get_token(
+                self.organization, f"{self.provider}_scim"
+            )
+        else:
+            logger.warning(
+                "SCIM disabled but tried to access token",
+                extra={"organization_id": self.organization.id},
+            )
+            return None
+
+    def get_scim_url(self):
+        if self.flags.scim_enabled:
+            return f"https://sentry.io/scim/{self.organization.slug}/scim/v2"
+            # TODO: make dynamic once URL routes are added
+        else:
+            return None
+
+    def enable_scim(self, user):
+        from sentry.mediators.sentry_apps import InternalCreator
+        from sentry.models import SentryAppInstallation, SentryAppInstallationForProvider
+
+        if (
+            not self.get_provider().can_use_scim(self.organization, user)
+            or self.flags.scim_enabled is True
+        ):
+            logger.warning(
+                "SCIM already enabled",
+                extra={"organization_id": self.organization.id},
+            )
+            return
+
+        # check if we have a scim app already
+
+        if SentryAppInstallationForProvider.objects.filter(
+            organization=self.organization, provider="okta_scim"
+        ).exists():
+            logger.warning(
+                "SCIM instillation already exists",
+                extra={"organization_id": self.organization.id},
+            )
+            return
+
+        data = {
+            "name": "SCIM Internal Integration",
+            "author": "Auto-generated by Sentry",
+            "organization": self.organization,
+            "overview": SCIM_INTERNAL_INTEGRATION_OVERVIEW,
+            "user": user,
+            "scopes": [
+                "member:read",
+                "member:write",
+                "member:admin",
+                "team:write",
+                "org:write",
+                "org:admin",
+            ],
+        }
+        # create the internal integration and link it to the join table
+        sentry_app = InternalCreator.run(**data)
+        sentry_app_installation = SentryAppInstallation.objects.get(sentry_app=sentry_app)
+        SentryAppInstallationForProvider.objects.create(
+            sentry_app_installation=sentry_app_installation,
+            organization=self.organization,
+            provider=f"{self.provider}_scim",
+        )
+        self.flags.scim_enabled = True
+
+    def disable_scim(self, user):
+        from sentry.mediators.sentry_apps import Destroyer
+        from sentry.models import SentryAppInstallationForProvider
+
+        if self.flags.scim_enabled:
+            install = SentryAppInstallationForProvider.objects.get(
+                organization=self.organization, provider=f"{self.provider}_scim"
+            )
+            Destroyer.run(sentry_app=install.sentry_app_installation.sentry_app, user=user)
+            self.flags.scim_enabled = False
```

Comment:
Should this be persisted at the same time as the install is destroyed? It looks like you could end up in a state where the installation is destroyed but the flag state is not persisted.